haproxy/tests/conftest.py

```diff
@@ -44,8 +44,16 @@ def dd_environment():
 
 @pytest.fixture(scope='session')
 def prometheus_metrics():
-    metrics = list(METRIC_MAP.values())
+    metrics = METRIC_MAP
```

Comment:
```suggestion
    from copy import deepcopy
    [...]
    metrics = deepcopy(METRIC_MAP)
```
We need to deepcopy before `pop`, or the `METRICS_MAP` will be updated as well

Reply:
ah thanks, now fixed!